tests/std/tests/P0980R1_constexpr_strings/test.cpp

```diff
@@ -0,0 +1,1647 @@
+    { // allocator
+        str default_constructed;
+        [[maybe_unused]] const auto alloc = default_constructed.get_allocator();
+        static_assert(is_same_v<remove_const_t<decltype(alloc)>, allocator<CharType>>);
+    }
+#if defined(MSVC_INTERNAL_TESTING) || defined(__EDG__) || _ITERATOR_DEBUG_LEVEL != 0 // TRANSITION, VSO-1269894
+    { // access
+        str literal_constructed             = get_literal_input<CharType>();
+        const str const_literal_constructed = get_literal_input<CharType>();
+
+        const auto at = literal_constructed.at(2);
+        static_assert(is_same_v<remove_const_t<decltype(at)>, CharType>);
+        assert(at == CharType{'l'});
+
+        literal_constructed.at(2) = CharType{'v'};
+
+        const auto at2 = literal_constructed.at(2);
+        static_assert(is_same_v<remove_const_t<decltype(at2)>, CharType>);
+        assert(at2 == CharType{'v'});
+
+        const auto cat = const_literal_constructed.at(2);
+        static_assert(is_same_v<remove_const_t<decltype(cat)>, CharType>);
+        assert(cat == CharType{'l'});
+
+        const auto op = literal_constructed[3];
+        static_assert(is_same_v<remove_const_t<decltype(op)>, CharType>);
+        assert(op == CharType{'l'});
+
+        literal_constructed[3] = CharType{'u'};
+        const auto op2         = literal_constructed[3];
+        static_assert(is_same_v<remove_const_t<decltype(op2)>, CharType>);
+        assert(op2 == CharType{'u'});
+
+        const auto cop = const_literal_constructed[3];
+        static_assert(is_same_v<remove_const_t<decltype(cop)>, CharType>);
+        assert(cop == CharType{'l'});
+
+        const auto f = literal_constructed.front();
+        static_assert(is_same_v<remove_const_t<decltype(f)>, CharType>);
+        assert(f == CharType{'H'});
+
+        const auto cf = const_literal_constructed.front();
+        static_assert(is_same_v<remove_const_t<decltype(cf)>, CharType>);
+        assert(cf == CharType{'H'});
+
+        const auto b = literal_constructed.back();
+        static_assert(is_same_v<remove_const_t<decltype(b)>, CharType>);
+        assert(b == CharType{'s'});
+
+        const auto cb = const_literal_constructed.back();
+        static_assert(is_same_v<remove_const_t<decltype(cb)>, CharType>);
+        assert(cb == CharType{'s'});
+
+        const auto d = literal_constructed.data();
+        static_assert(is_same_v<decltype(d), CharType* const>);
+        assert(*d == CharType{'H'});
+
+        const auto cd = const_literal_constructed.data();
+        static_assert(is_same_v<decltype(cd), const CharType* const>);
+        assert(*cd == CharType{'H'});
+
+        const auto cs = literal_constructed.c_str();
+        static_assert(is_same_v<decltype(d), CharType* const>);
+        assert(cs == literal_constructed.data());
+        assert(char_traits<CharType>::length(cs) == literal_constructed.size());
+    }

```

Comment:
I think we're also not exercising `operator basic_string_view<charT, traits>()` anywhere - we should add a brief call to this function.